Zend/zend_vm_def.h

```diff
@@ -9380,6 +9380,130 @@ ZEND_VM_HANDLER(202, ZEND_CALLABLE_CONVERT, UNUSED, UNUSED)
 	ZEND_VM_NEXT_OPCODE();
 }
 
+ZEND_VM_HANDLER(203, ZEND_INIT_OBJ, VAR|UNUSED|THIS|CV, CONST, CACHE_SLOT, SPEC(OP_DATA=CONST|TMP|VAR|CV))
+{
+	USE_OPLINE
+	zval *object, *value, tmp;
+	zend_object *zobj;
+	zend_string *name;
+
+	SAVE_OPLINE();
+	object = GET_OP1_OBJ_ZVAL_PTR_PTR_UNDEF(BP_VAR_W);
+	value = GET_OP_DATA_ZVAL_PTR(BP_VAR_R);
+
+	if (OP1_TYPE != IS_UNUSED && UNEXPECTED(Z_TYPE_P(object) != IS_OBJECT)) {
+		if (Z_ISREF_P(object) && Z_TYPE_P(Z_REFVAL_P(object)) == IS_OBJECT) {
+			object = Z_REFVAL_P(object);
+			ZEND_VM_C_GOTO(init_object);
+		}
+		zend_throw_non_object_error(object, GET_OP2_ZVAL_PTR(BP_VAR_R) OPLINE_CC EXECUTE_DATA_CC);
+		value = &EG(uninitialized_zval);
+		ZEND_VM_C_GOTO(free_and_exit_init_obj);
+	}
+
+ZEND_VM_C_LABEL(init_object):
+	zobj = Z_OBJ_P(object);
+	void **cache_slot = CACHE_ADDR(opline->extended_value);
+	if (EXPECTED(zobj->ce == CACHED_PTR(opline->extended_value))) {
```

Comment:
`if (EXPECTED(zobj->ce == CACHED_PTR(opline->extended_value))) {` - This code path will only be reached the **second** time you execute code.

So for many of the tests, they should be run in a for loop to run twice, to check that they do the same thing twice in a row.

The `--repeat` option won't help, since I'm pretty sure the run time cache is distinct for separate php requests

It still makes sense to have, since the optimized code will be faster in applications that call a given php `clone with` expression multiple times

Reply:
Hm, yes, good question! I'm not yet sure... I guess if the opcode is not used often then the gain with the cache is not much. In the same time, if the cache is not used much then it doesn't cause a big pressure (?). Anyway, I'm open to remove this path (given the RFC passes) :)